Components/Core/Timer.cpp

```diff
@@ -21,14 +22,179 @@ Timer::Timer()
 	// The timer ID is specified as (void *)this to provide a unique ID for each timer object - however this is not necessary for polling timers.
 	// The timer is created in the dormant state.
 	rtTimerHandle = xTimerCreate("Timer", DEFAULT_TIMER_PERIOD, pdFALSE, (void *)this, empty_callback);
+	SOAR_ASSERT(rtTimerHandle, "Error Occurred, Timer not created");
+	timerState = UNINITIALIZED;
+}
+
+/**
+ * @brief Default de-constructor makes a timer that can only be polled for state
+*/
+
+Timer::~Timer()
+{
+	xTimerDelete(rtTimerHandle,(DEFAULT_TIMER_COMMAND_WAIT_PERIOD*2));
+	SOAR_ASSERT(rtTimerHandle, "Error Occurred, Timer could not be deleted");
```

Comment:
1. Please change this to check the output of xTimerDelete instead of rtTimerHandle, this doesn't check for much other than rtTimerHandle != null.
2. Please change to an if statement and SOAR_PRINT instead, it isn't so severe that we need to reset the system over a timer not getting deleted.